src/integration-tests/src/workflows.rs

```diff
@@ -0,0 +1,255 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Result;
+use gax::error::Error;
+use lro::Poller;
+use rand::{distributions::Alphanumeric, Rng};
+use std::time::Duration;
+
+pub const WORKFLOW_ID_LENGTH: usize = 64;
+
+pub async fn run(config: Option<gax::options::ClientConfig>) -> Result<()> {
+    // Enable a basic subscriber. Useful to troubleshoot problems and visually
+    // verify tracing is doing something.
+    #[cfg(feature = "log-integration-tests")]
+    let _guard = {
+        use tracing_subscriber::fmt::format::FmtSpan;
+        let subscriber = tracing_subscriber::fmt()
+            .with_level(true)
+            .with_thread_ids(true)
+            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+            .finish();
+
+        tracing::subscriber::set_default(subscriber)
+    };
+
+    let project_id = crate::project_id()?;
+    let location_id = crate::region_id();
+
+    // We could simplify the code, but we want to test both ::new_with_config()
+    // and ::new().
+    let client = if let Some(config) = config {
+        wf::client::Workflows::new_with_config(config).await?
+    } else {
+        wf::client::Workflows::new().await?
+    };
+    cleanup_stale_workflows(&client, &project_id, &location_id).await?;
+
+    let source_contents = r###"# Test only workflow
+main:
+    steps:
+        - sayHello:
+            return: Hello World
+"###;
+    let source_code = wf::model::workflow::SourceCode::SourceContents(source_contents.to_string());
+
+    let workflow_id: String = rand::thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(WORKFLOW_ID_LENGTH - 4)
```

Comment:
Why `-4`? should the ID len just be that value?

Reply:
Made the comments more clear (I hope).